### DISCUSSION TITLE
Overly verbose documentation overlay?

### DISCUSSION THREAD

**Discussion by dandy-step** [kitamstudios/rust-analyzer.vs#42] [Q&A] [answered]
Likely a problem with my configuration, but the extension is borderline unusable for me, since it immediately and automatically displays a huge tooltip with documentation that covers up the actual code and doesn't go away unless I move the mouse away from the box or repeatedly hit Esc, whenever I mouse over anything that it has docs for:

![image](https://github.com/kitamstudios/rust-analyzer.vs/assets/29162970/9852233f-b5f0-49fb-89ea-080d55c71650)

The documentation is helpful, but is there a way to trim or limit the amount of information presented, or configure the delay so that it's not as immediate?

**Comment by parthopdas** [comment, marked as answer]
Yep, known issues. Its on the edges of publicly available documentation. I have reached out to Microsoft for help.

**Reply by dandy-step** [reply]
Thank you for your work, and please let me know if there's anything I can do to help or somehow raise awareness for this issue.

**Comment by opsidjflksdf** [comment]
I personally strongly prefer not to have any tooltips shown at all. 
For C#, I found an extension that works to completely prevent tooltips from being shown.
https://github.com/gs412/Tame-Visual-Studio-Code-Editor-Tooltips/tree/master/TameVisualStudioEditorToolTips

It's pretty short code, I listed it below.
I wonder if the same type of code could work for Rust. 
For example if the rust_analyzer extension would provide the infrastructure to be able to use an attribute like
```[ContentType("Rust")]``` in addition to ```[ContentType("CSharp")]```

If we're able to hide tooltips in Rust, that could also be a good interim for you guys till the overlay text could be cut down or other options tweaked.

```
namespace TameVisualStudioEditorToolTips {

    using System.ComponentModel.Composition;
    using System.Threading;
    using System.Threading.Tasks;
    using System.Windows.Input;
    using Microsoft.VisualStudio.Language.Intellisense;
    using Microsoft.VisualStudio.Shell;
    using Microsoft.VisualStudio.Text;
    using Microsoft.VisualStudio.Utilities;

    [Export(typeof(IAsyncQuickInfoSourceProvider))]
    [Name("Tame Quick Info")]
    [Order(Before = "Default Quick Info Presenter")]
    [ContentType("CSharp")]
    [ContentType("XAML")]
    [ContentType("XML")]
    internal class TameQuickInfo : IAsyncQuickInfoSourceProvider {

        public IAsyncQuickInfoSource TryCreateQuickInfoSource(ITextBuffer textBuffer) {
            return textBuffer.Properties.GetOrCreateSingletonProperty(() => new CancelingQuickInfoSource());
        }
    }

    internal class CancelingQuickInfoSource : IAsyncQuickInfoSource {

        public CancelingQuickInfoSource() {
        }

        public void Dispose() {
        }

        public async Task<QuickInfoItem> GetQuickInfoItemAsync(IAsyncQuickInfoSession session, CancellationToken cancellationToken)
        {
            
            await session.DismissAsync();
            
            return await System.Threading.Tasks.Task.FromResult<QuickInfoItem>(null);
        }
    }
}
```
